src/AppBundle/Service/JailService.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace AppBundle\Service;
+
+class JailService
+{
+    private $geoLocation;
+    private $logger;
+
+    public function __construct(GeoLocation $geoLocation, LogService $logger)
+    {
+        $this->geoLocation = $geoLocation;
+        $this->logger = $logger;
+    }
+
+    public function banCurrentIpIfForeign()
+    {
+        $ip = $this->geoLocation->clientIp();
+        $countryCode = $this->geoLocation->getCurrentCountryCode();
+        $currentIpIsForeign = $ip !== null && $countryCode !== null && $countryCode !== 'NO';
+
+        if ($currentIpIsForeign) {
+            $this->banIp($ip);
+        }
+    }
+
+    private function banIp(string $ip)
+    {
+        $countryCode = $this->geoLocation->getCurrentCountryCode();
+        shell_exec("sudo fail2ban-client set vektorhell banip $ip");
```

Comment:
Jeg synes det virker litt spooky å hente ut ipen fra en request header og deretter bruke den i `shell_exec`. Hva tenker du om dette? Er vi helt sikre på at det ikke er mulig med et injection-angrep her?

Reply:
Ja, tenker å hente IP ut fra Request-objektet fra Symfony i JailService og GeoLocation. Og bytte ut shell_exec med Symfony.

> The Process class executes a command in a sub-process, taking care of the differences between operating system and escaping arguments to prevent security issues.

Reply:
Ah sweet, da blir det nok sikkert :+1: